draft-chen-cfrg-vdaf-pine.md

```diff
@@ -346,10 +347,80 @@ implements this encoding step.
 
 ## The FLP Circuit
 
-> XXX Give an overview of the circuit's operation. Here it would be appropriate
-> to name all of the intermediate results: the bit checks, the wraparound
-> checks result (the reduction over all of the checks), the wraparound success
-> count, the squred norm equality test, and the squared norm range check.
+The validity circuit unpacks the encoded measurement as the following:
+
+* The first `dimension` entries are the `encoded_gradient`, the field elements
+  encoded from the floating point numbers.
+* The next `bit_checked_len` entries are expected to be bits, and should contain
+  the bits from L2-norm check, the bits from the range check in wraparound
+  checks, and the success bits in wraparound checks.
+* The last `num_wr_checks` are the wraparound check results, i.e., the dot
+  products of the encoded gradient and the random vectors.
+
+It also unpacks the joint randomness to compute random linear combinations of
+all the checks:
+
+* The first joint randomness field element is to reduce over the bit checks at
+  all bit entries.
+* The second joint randomness field element is to reduce over all the quadratic
+  checks in wraparound check.
+* The last joint randomness field element is to reduce over all the checks,
+  which include the reduced bit check result, the L2-norm equality check, the
+  L2-norm range check, the reduced quadratic checks in wraparound check, and
+  the success count check for wraparound check.
+
+Some of the auxiliary functions are defined in {{pine-auxiliary}}.
+
+### Bit Check {#valid-bit-check}
+
+The bit check computes a random linear combination of the polynomial `x(x-1)`
+evaluated at all bit entries of `bit_checked`, with the coefficients computed
+from the joint randomness `r_bit_check`. Note we need to normalize the constant
+`1` in the polynomial, by multiplying it by the inverse of number of shares
+`shares_inv`.
```

Comment:
I'd say let's save this note until we've handled the TODO below.

In fact, it may be worth adding an in line comment somewhere in `eval()` explaining the purpose of `shares_inv`.